### PR TITLE
Update Elixir to v12 in CI

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
This commit updates Elixir to v12 in github actions

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
         uses: erlef/setup-elixir@v1
         with:
           otp-version: 23.1.1
-          elixir-version: 1.11.0
+          elixir-version: 1.12.0
```
Comment (wojtekmach) [inline .github/workflows/main.yml:16]:
```suggestion
          elixir-version: 1.12
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ jobs:
...
```
Comment (wojtekmach) [inline .github/workflows/main.yml:45]:
```suggestion
          elixir-version: 1.12
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -86,7 +86,7 @@ jobs:
...
```
Comment (wojtekmach) [inline .github/workflows/main.yml:89]:
```suggestion
          elixir-version: 1.12
```

**Comment by wojtekmach** [comment]
Thanks!

File: .github/workflows/main.yml (hunk 1 above)
Comment (ericmj) [reply .github/workflows/main.yml]: We should depend on a single version of Elixir to prevent potential future breakage.

**Comment by ericmj** [comment]
We also need to update the Dockerfile version: https://github.com/hexpm/hexpm/blob/main/Dockerfile#L1.

**Pull request merged**